src/java.base/share/classes/jdk/internal/util/OperatingSystem.java

```diff
@@ -129,7 +127,34 @@ public static OperatingSystem current() {
      * Names not recognized throw ExceptionInInitializerError with IllegalArgumentException.
      */
     private static OperatingSystem initOS(String osName) {
-        return OperatingSystem.valueOf(osName.toUpperCase(Locale.ROOT));
+        // Too early to use Locale conversions, manually do uppercase
+        StringBuilder sb = new StringBuilder(osName);
+        for (int i = 0; i < sb.length(); i++) {
+            char ch = sb.charAt(i);
+            if (ch >= 'a' && ch <= 'z') {
+                sb.setCharAt(i, (char)(ch - ('a' - 'A')));  // Map lower case down to uppercase
+            }
+        }
+        osName = sb.toString();
+        return OperatingSystem.valueOf(osName);
     }
 
+    /**
+     * {@return the operating system version with major, minor, micro}
+     */
+    public static Version version() {
+        return osVersion;
+    }
+
+    // Parse and save the current version
+    private static Version osVersion = initVersion();
```

Comment:
The version field should probably be `final`:
```suggestion
    public static Version version() {
        return CURRENT_VERSION;
    }

    // Parse and save the current version
    private static final Version CURRENT_VERSION = initVersion();
```